app/models/assignment.rb

```diff
@@ -1156,6 +1156,31 @@ def to_json(options = {})
     self.assignment_properties.attributes.merge(self.attributes).symbolize_keys.to_json(options)
   end
 
+  # Returns an assignment's relevant properties for uploading/downloading an assignment's configuration as a hash
+  def assignment_properties_config
+    properties = {}
+    # Insert DEFAULT_FIELDS first to make file more familiar
+    DEFAULT_FIELDS.each do |f|
+      properties[f] = self.assignment.send(f)
+    end
+    # Get all other assignment properties
+    other_properties = self.assignment_properties.attributes.merge(self.attributes).symbolize_keys
```

Comment:
I don't love the idea of just grabbing all remaining keys from assignment_properties. At the very least I would prefer to have which keys you want listed somewhere. Either update `DEFAULT_FIELDS` to include the other keys you want to grab or create another global variable similar to `DEFAULT_FIELDS` that includes the rest of the keys you want to grab and use that.